File: PDOO/Material Complementario/Tema 1/Peluqueria/src/basico/Peluqueria.java

package basico;

import java.time.LocalDateTime;
import java.time.Month;
import java.util.ArrayList;

public class Peluqueria {  //Clase con visibilidad pública. Su constructor es el que ya existe por defecto
    
    ArrayList<Cita> citas= new ArrayList<>();
       
    Cita darCita(Persona persona){ //Da una cita en un día y hora aleatorios
       	int year= (int)(Math.random()*2+2020);	//hasta 2 años a partir del 2020
        int dia= (int) (Math.random()*30+1);
        int hora= (int) (Math.random()*8+9);      
        LocalDateTime fecha= LocalDateTime.of(year, Month.OCTOBER, dia, hora, 0);
        Cita cita = new Cita(fecha,persona);
        citas.add(cita);
        System.out.println(cita.getFecha());
        return cita;
    }
    
    void cambiarColorPelo(Persona p, ColorPelo colorNuevo){
        p.pelo = colorNuevo;
    }
    
    void anularCita(Cita borrar){
        System.out.println("Peluqueria: cita anulada");
        citas.remove(borrar);
    }
    
}

File: PDOO/Material Complementario/Tema 1/PeluqueriaRuby/lib/peluqueria.rb
module Basico

  require_relative "cita"
  
  class Peluqueria
    
  def initialize
    @citas=Array.new
  end
  
  def dar_cita(persona)
    year= 2020+rand(3) #hasta 2 años a partir de 2020
    dia= 1+ rand(31) 
    hora= 9+rand(9)
    fecha= Time.new(year, 11, dia, hora)
    cita = Cita.new(persona, fecha)
    @citas<<cita
    cita
  end
  
  def poner_rubio(p)
    p.pelo=ColorPelo::RUBIO
  end
  
  def anular_cita(cita)
    @citas.delete(cita)
    puts "Cita anulada"
  end
end

end
